### PR TITLE
Feature/hierachical approve

### PR DESCRIPTION
**Pull request by yunhangc**

### CODE REVIEW

File: rbac/processor/role/role_operation.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Copyright 2017 Intel Corporation
```
Comment (chrisspanton) [inline rbac/processor/role/role_operation.py]: Nitpick - please use latest Copyright line. "Copyright 2018 Contributors to Hyperledger Sawtooth"

Comment (kthblmfld) [inline rbac/processor/role/role_operation.py]: Please update this header

File: rbac/processor/role/role_validator.py [hunk 2]
```diff
@@ -195,3 +195,61 @@ def _role_already_exists(state_return, role_id):
     return message_accessor.is_in_role_attributes_container(
         container=role_attr_container, identifier=role_id
     )
+
+def get_state_entries(header, confirm, txn_signer_rel_address, state):
```
Comment (kthblmfld) [inline rbac/processor/role/role_validator.py]: Why move this from state_accessor? It looks like the main goal of the function is to retrieve state entries from the given confirmation.

File: rbac/processor/role/role_admins.py [hunk 3]
```diff
@@ -166,3 +168,18 @@ def apply_reject_remove(header, payload, state):
             header, payload, state
         )
     )
+def hierachical_approve(header, payload, state):
```
Comment (chrisspanton) [inline rbac/processor/role/role_admins.py]: Consistent but incorrect spelling across this commit for "hierarchical". Please address.

File: rbac/processor/role/role_members.py [hunk 4]
```diff
@@ -103,6 +106,21 @@ def apply_propose_remove(header, payload, state):
         state=state,
     )
 
+def hierachical_approve(header, payload, state):
```
Comment (chrisspanton) [inline rbac/processor/role/role_members.py]: "hierarchical"

File: rbac/processor/role/role_owners.py [hunk 5]
```diff
@@ -103,6 +106,21 @@ def apply_propose_remove(header, payload, state):
         state=state,
     )
 
+def hierachical_approve(header, payload, state):
```
Comment (chrisspanton) [inline rbac/processor/role/role_owners.py]: "hierarchical"

File: rbac/processor/role/role_members.py (hunk 4 above)
Comment (chrisspanton) [reply rbac/processor/role/role_members.py]: Please update across all instances in all files in this commit. It seems mostly consistent, but I did see at least one example of spelling it appropriately. Bigger picture the question is, whether this should be changed altogether, but lets fix and merge this feature for now.

File: rbac/processor/role/role_operation.py (hunk 1 above)
Comment (chrisspanton) [reply rbac/processor/role/role_operation.py]: first... ;)

File: rbac/processor/role/role_admins.py (hunk 3 above)
Comment (yunhangc) [reply rbac/processor/role/role_admins.py]: haha. english is not my first language. will fix. thanks for pointing out

File: rbac/processor/role/role_operation.py (hunk 1 above)
Comment (yunhangc) [reply rbac/processor/role/role_operation.py]: will fix

**Comment by yunhangc** [comment]
@dcmiddle Dan, your comment is resolved

**Pull request merged**